pkg/apis/pipeline/v1/pipeline_types_test.go

```diff
@@ -0,0 +1,1048 @@
+			}
+		})
+	}
+}
+
+func TestPipelineTask_Validate_Failure(t *testing.T) {
+	tests := []struct {
+		name          string
+		p             PipelineTask
+		expectedError apis.FieldError
+		wc            func(context.Context) context.Context
+	}{{
+		name: "invalid custom task without Kind",
+		p: PipelineTask{
+			Name:    "invalid-custom-task",
+			TaskRef: &TaskRef{APIVersion: "example.com"},
+		},
+		expectedError: apis.FieldError{
+			Message: `invalid value: custom task ref must specify kind`,
+			Paths:   []string{"taskRef.kind"},
+		},
+		wc: enableFeatures(t, []string{"enable-custom-tasks"}),
+	}, {
+		name: "invalid bundle without bundle name",
+		p: PipelineTask{
+			Name:    "invalid-bundle",
+			TaskRef: &TaskRef{Bundle: "bundle"},
+		},
+		expectedError: apis.FieldError{
+			Message: `missing field(s)`,
+			Paths:   []string{"taskRef.name"},
+		},
+		wc: enableFeatures(t, []string{"enable-tekton-oci-bundles"}),
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			if tt.wc != nil {
+				ctx = tt.wc(ctx)
+			}
+			err := tt.p.Validate(ctx)
+			if err == nil {
+				t.Error("PipelineTask.Validate() did not return error for invalid pipeline task")
+			}
+			if d := cmp.Diff(tt.expectedError.Error(), err.Error(), cmpopts.IgnoreUnexported(apis.FieldError{})); d != "" {
+				t.Errorf("PipelineTask.Validate() errors diff %s", diff.PrintWantGot(d))
+			}
+		})
+	}
+}
+
+func TestPipelineTaskList_Names(t *testing.T) {
+	tasks := []PipelineTask{
+		{Name: "task-1"},
+		{Name: "task-2"},
+	}
+	expectedTaskNames := sets.String{}
+	expectedTaskNames.Insert("task-1")
+	expectedTaskNames.Insert("task-2")
+	actualTaskNames := PipelineTaskList(tasks).Names()
+	if d := cmp.Diff(expectedTaskNames, actualTaskNames); d != "" {
+		t.Fatalf("Failed to get list of pipeline task names, diff: %s", diff.PrintWantGot(d))
+	}
+}
+
+func TestPipelineTaskList_Deps(t *testing.T) {
+	pipelines := []struct {
+		name         string
+		tasks        PipelineTaskList
+		expectedDeps map[string][]string
+	}{{
+		name: "valid pipeline without any deps",
+		tasks: []PipelineTask{
+			{Name: "task-1"},
+			{Name: "task-2"},
+		},
+		expectedDeps: map[string][]string{},
+	}, {
+		name: "valid pipeline with ordering deps - runAfter",
+		tasks: []PipelineTask{
+			{Name: "task-1"},
+			{Name: "task-2", RunAfter: []string{"task-1"}},
+		},
+		expectedDeps: map[string][]string{
+			"task-2": {"task-1"},
+		},
+	}, {
+		name: "valid pipeline with resource deps - Inputs",
+		tasks: []PipelineTask{{
+			Name: "task-1",
+		}, {
+			Name: "task-2",
+		}, {
+			Name: "task-3",
+			Resources: &PipelineTaskResources{
+				Inputs: []PipelineTaskInputResource{{
+					From: []string{"task-1", "task-2"},
+				}},
+			}},
+		},
+		expectedDeps: map[string][]string{
+			"task-3": {"task-1", "task-2"},
+		},
+	}, {
+		name: "valid pipeline with resource deps - Task Results",
```

Comment:
Can I confirm that theses should be deleted as resources are deprecated?
(Keeping the when expression cases.)

Reply:
PipelineResources are deprecated, but results are not, and it looks like this test case is just using results. it should probably be renamed.

Reply:
Thanks Lee! Renamed at https://github.com/tektoncd/pipeline/pull/5228.